core/trino-main/src/test/java/io/trino/operator/TestDeduplicationExchangeClientBuffer.java

```diff
@@ -0,0 +1,443 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.operator;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Multimap;
+import com.google.common.collect.Sets;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.units.DataSize;
+import io.trino.execution.StageId;
+import io.trino.execution.TaskId;
+import io.trino.execution.buffer.PageCodecMarker;
+import io.trino.execution.buffer.SerializedPage;
+import io.trino.spi.TrinoException;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.util.concurrent.MoreExecutors.directExecutor;
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.airlift.units.DataSize.Unit.BYTE;
+import static io.airlift.units.DataSize.Unit.KILOBYTE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+public class TestDeduplicationExchangeClientBuffer
+{
+    private static final DataSize ONE_KB = DataSize.of(1, KILOBYTE);
+
+    @Test
+    public void testIsBlocked()
+    {
+        try (ExchangeClientBuffer buffer = new DeduplicationExchangeClientBuffer(directExecutor(), ONE_KB, RetryPolicy.QUERY)) {
+            ListenableFuture<Void> blocked = buffer.isBlocked();
+            assertFalse(blocked.isDone());
+            buffer.close();
+            assertTrue(blocked.isDone());
+        }
+
+        try (ExchangeClientBuffer buffer = new DeduplicationExchangeClientBuffer(directExecutor(), ONE_KB, RetryPolicy.QUERY)) {
+            ListenableFuture<Void> blocked = buffer.isBlocked();
+            assertFalse(blocked.isDone());
```

Comment:
nit: can you extract `assertBlocked(ListenableFuture)` and `assertNotBlocked` helper methods. 
mapping `assertFalse` to positive "blocked" and `assertTrue` to negative "not-blocked" makes it hard to read.